core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.controller.ts

```diff
@@ -120,122 +156,147 @@ angular.module('oppia').directive('topicsAndSkillsDashboardPage', [
-                    newExplanationObject = explanationObject.toBackendDict();
-                    $scope.bindableDict.displayedConceptCardExplanation =
-                      explanationObject.getHtml();
-                  };
-
-                  $scope.onSaveRubric = function(difficulty, explanations) {
-                    for (var idx in $scope.rubrics) {
-                      if ($scope.rubrics[idx].getDifficulty() === difficulty) {
-                        $scope.rubrics[idx].setExplanations(explanations);
-                      }
-                    }
-                  };
-
-                  $scope.resetErrorMsg = function() {
-                    $scope.errorMsg = '';
-                  };
-
-                  $scope.createNewSkill = function() {
-                    if (
-                      !SkillObjectFactory.hasValidDescription(
-                        $scope.newSkillDescription)) {
-                      $scope.errorMsg = (
-                        'Please use a non-empty description consisting of ' +
-                        'alphanumeric characters, spaces and/or hyphens.');
-                      return;
-                    }
-                    $uibModalInstance.close({
-                      description: $scope.newSkillDescription,
-                      rubrics: $scope.rubrics,
-                      explanation: newExplanationObject
-                    });
-                  };
-
-                  $scope.cancel = function() {
-                    ImageLocalStorageService.flushStoredImagesData();
-                    SkillCreationService.resetSkillDescriptionStatusMarker();
-                    $uibModalInstance.dismiss('cancel');
-                  };
-                }
-              ]
+              controller: 'CreateNewSkillModalController'
             }).result.then(function(result) {
               ContextService.resetImageSaveDestination();
               SkillCreationService.createNewSkill(
                 result.description, result.rubrics, result.explanation, []);
             });
           };
+          /**
+           * @param {Number} pageNumber - Page number to navigate to.
+           */
+          ctrl.goToPageNumber = function(pageNumber) {
+            if (ctrl.activeTab === ctrl.TAB_NAME_TOPICS) {
+              ctrl.topicPageNumber = pageNumber;
+              ctrl.pageNumber = ctrl.topicPageNumber;
+              ctrl.displayedTopicSummaries =
+                  ctrl.topicSummaries.slice(
+                    pageNumber * ctrl.itemsPerPage,
+                    (pageNumber + 1) * ctrl.itemsPerPage);
+            } else if (ctrl.activeTab === ctrl.TAB_NAME_UNTRIAGED_SKILLS) {
+              ctrl.totalEntityCountToDisplay =
+                  ctrl.totalUntriagedSkillSummaries.length;
+              ctrl.skillPageNumber = pageNumber;
+              ctrl.pageNumber = ctrl.skillPageNumber;
+              ctrl.untriagedSkillSummaries =
```

Comment:
Wrap in `(..)` when you have statements spanning multiple lines.

Reply:
Done